### PR TITLE
Allow duplicate column names in Joiner

### PR DESCRIPTION
**Pull request by jeromedockes**
random (but consistent across calls to transform) strings are inserted in duplicate names to avoid errors.
This is to be consistent with the AggJoiner and AggTarget

### CODE REVIEW

**Review by Vincent-Maladiere — APPROVED** [review]
Very nice follow-up, LGTM

File: skrub/_joiner.py [hunk 1]
```diff
@@ -294,10 +294,9 @@ def fit(self, X, y=None):
 
         Returns
         -------
-        Joiner
-            Fitted Joiner instance (self).
+        dataframe
```
Comment (TheooJ) [inline skrub/_joiner.py:297]:
```suggestion
        DataFrame
```

File: skrub/_joiner.py [hunk 2]
```diff
@@ -321,6 +317,26 @@ def fit(self, X, y=None):
...
+        ----------
+        X : dataframe
+            The main table, to be joined to the auxiliary ones.
+        y : None
+            Unused, only here for compatibility.
+
+        Returns
+        -------
+        Joiner
+            Fitted Joiner instance (self).
```
Comment (TheooJ) [inline skrub/_joiner.py:337]:
```suggestion
            Fitted :class:`Joiner`instance (self).
```

File: skrub/_joiner.py [hunk 3]
```diff
@@ -321,6 +317,26 @@ def fit(self, X, y=None):
...
+        result = self._transform(X, y)
+        self.all_outputs_ = sbd.column_names(result)
+        return result
+
+    def fit(self, X, y=None):
+        """Fit the instance to the main table.
+
+        Parameters
+        ----------
+        X : dataframe
```
Comment (TheooJ) [inline skrub/_joiner.py:329]:
```suggestion
        X : DataFrame
```

**Review by TheooJ — APPROVED** [review]
Thanks, just some nitpicks

Do we want an entry in the docstrings `Attributes` section for `all_outputs_` ?

**Comment by TheooJ** [comment]
Thanks @jeromedockes !

**Pull request merged**